src/transformers/models/siglip/modeling_siglip.py

```diff
@@ -1018,6 +1018,13 @@ def forward(
 @add_start_docstrings(SIGLIP_START_DOCSTRING)
 class SiglipModel(SiglipPreTrainedModel):
     config_class = SiglipConfig
+    _no_split_modules = [
+        "SiglipTextEmbeddings",
+        "SiglipEncoderLayer",
+        "SiglipVisionEmbeddings",
+        "SiglipEncoderLayer",
+        "SiglipMultiheadAttentionPoolingHead",
+    ]
```

Comment:
What happens if we have modules defined in `_no_split_modules`  which aren't in the model? It probably makes more sense to move this into the `PretrainedModel` definition than repeat e.g. `"SiglipVisionEmbeddings"` for all models which use the vision components

Reply:
Great idea! Didn't think that extra `split_modules` will not affect anything. Moved to SiglipPretrained model and ran tests. Had to skip offloading, same way it's skipped in SigLipVision